rest_framework_json_api/renderers.py

```diff
@@ -244,14 +256,23 @@ def extract_relationships(cls, fields, resource, resource_instance):
                 if isinstance(serializer_data, list):
                     for position in range(len(serializer_data)):
                         nested_resource_instance = resource_instance_queryset[position]
+                        nested_resource_data = serializer_data[position]
                         nested_resource_instance_type = (
                             relation_type or
                             utils.get_resource_type_from_instance(nested_resource_instance)
                         )
 
+                        instance_pk_name = (
+                            getattr(field, 'lookup_field', None) or
+                            nested_resource_instance._meta.pk.name)
+                        if instance_pk_name in nested_resource_data:
+                            pk = nested_resource_data[instance_pk_name]
```

Comment:
Could you change this to use a consistent name? Higher in this diff, the variable passed to `force_text` was `_id`. Now it's `pk`. I think we should pick one and stick with it.